packages/core/src/shared/filetypes.ts

```diff
@@ -353,8 +353,19 @@ export const codefileExtensions = new Set([
 // Code file names without an extension
 export const codefileNames = new Set(['Dockerfile', 'Dockerfile.build'])
 
+// To match gradlew, .mvn/*, etc
+export const wellKnownSubnames = new Set(['gradle', 'mvn'])
+
 /** Returns true if `filename` is a code file. */
 export function isCodeFile(filename: string): boolean {
     const ext = path.extname(filename).toLowerCase()
-    return codefileExtensions.has(ext) || codefileNames.has(path.basename(filename))
+    if (codefileExtensions.has(ext) || codefileNames.has(path.basename(filename))) {
+        return true
+    }
+    for (const subname of wellKnownSubnames) {
+        if (filename.includes(subname)) {
```

Comment:
this is a very loose check.

 surely these changes deserve a test...

Reply:
this implicitness was intentional, as I wouldn't want to chase down specific extensions/file names for them. but we could also just include the missing file names and extensions. updated to the latter and added some tests.